src/util.ts

```diff
@@ -498,18 +495,18 @@ export function toTypedArray(
   if (dtype === 'string') {
     throw new Error('Cannot convert a string[] to a TypedArray');
   }
-  if (noConversionNeeded(a, dtype)) {
-    return a as TypedArray;
-  }
   if (Array.isArray(a)) {
     a = flatten(a as number[]);
   }
+  if (debugMode) {
```

Comment:
Seems like you fixed a bug where by moving this check sooner. Can you add a unit test to reproduce this (create a tensor directly from a TypedArray with NaNs in debug mode and assert that it throws)

Reply:
Done